From JBeulich@suse.com  Wed Mar  7 11:32:41 2012
From: "Jan Beulich" <JBeulich@suse.com>
Date: Tue, 28 Feb 2012 10:41:37 +0000
Subject: kprobes: adjust "fix a memory leak in function pre_handler_kretprobe()"
To: <stable@kernel.org>
Cc: <liuj97@gmail.com>,<gregkh@linuxfoundation.org>, "Jiri Slaby" <jslaby@suse.cz>
Message-ID: <4F4CBD710200007800075270@nat28.tlf.novell.com>
Content-Disposition: inline


From: Jan Beulich <JBeulich@suse.com>

3.0.21's 603b63484725a6e88e4ae5da58716efd88154b1e directly used
the upstream patch, yet kprobes locking in 3.0.x uses spin_lock...()
rather than raw_spin_lock...().

Signed-off-by: Jan Beulich <jbeulich@suse.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

---
 kernel/kprobes.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/kernel/kprobes.c
+++ b/kernel/kprobes.c
@@ -1663,9 +1663,9 @@ static int __kprobes pre_handler_kretpro
 		ri->task = current;
 
 		if (rp->entry_handler && rp->entry_handler(ri, regs)) {
-			raw_spin_lock_irqsave(&rp->lock, flags);
+			spin_lock_irqsave(&rp->lock, flags);
 			hlist_add_head(&ri->hlist, &rp->free_instances);
-			raw_spin_unlock_irqrestore(&rp->lock, flags);
+			spin_unlock_irqrestore(&rp->lock, flags);
 			return 0;
 		}
 
